### ISSUE TITLE
`prefer-for-of` false positive when iterator is in a callback function

### ISSUE THREAD

**Issue by nchen63** [palantir/tslint#2159]
### Bug Report

- __TSLint version__: ` 3.4.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: (please choose one) **CLI** / Node.js API / grunt-tslint / Atom / Visual Studio / etc

#### TypeScript code being linted
```
const columns = [{ data: [1] }];
for (let i = 0; i < columns[0].data.length; i++) {
    columns.map((x) => x.data[i]);
}
```
